pkg/disk/partition_table.go

```diff
@@ -894,3 +894,145 @@ func (pt *PartitionTable) GetMountpointSize(mountpoint string) (uint64, error) {
 
 	panic(fmt.Sprintf("no sizeable of the entity path for mountpoint %s, this is a programming error", mountpoint))
 }
+
+// EnsureRootFilesystem adds a root filesystem if the partition table doesn't
+// already have one.
+//
+// When adding the root filesystem, add it to:
+//
+//   - The first LVM Volume Group if one exists, otherwise
+//   - The first Btrfs volume if one exists, otherwise
+//   - At the end of the plain partitions.
+//
+// For LVM and Plain, the fsType argument must be a valid filesystem type.
+func EnsureRootFilesystem(pt *PartitionTable, fsType FSType) error {
+	// collect all labels and subvolume names to avoid conflicts
```

Comment:
We already have `entityPath()` which we use in a lot of places for this exact purpose, for example:
https://github.com/osbuild/images/blob/d9ffdd686ff3307d84c36b566dc48cacc37ca393/pkg/disk/partition_table.go#L468-L473

In fact, the first version of this function was doing just that before I realised we need to collect labels: https://github.com/osbuild/images/pull/926/commits/462ceca5f17f2e3045e476aee4b7d43086571da9#diff-ffc14a35d85eb6fd9724427b51786e40bc3e910bc41cbd6762ab977ae9689765R1170

Using that when we're going to scan through the PT anyway to collect the labels feels a bit wrong.  That said, one thing I noticed now is that we don't need to use the bool and can just return.

Reply:
Yeah, it's fine. I thought this was why it was done - to avoid the double traversal. I (personally) would still have made the tradeoff to do it twice and have a helper so that it can return early from a helper (as it feels easier for a human to read) but fine to leave as is, it is probably just personal.